idaes/power_generation/costing/tests/test_power_plant_costing.py

```diff
@@ -0,0 +1,555 @@
+        m.fs.boiler.costing.total_plant_cost['2.2']), abs=1e-1) == 13976/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.boiler.costing.total_plant_cost['4.11']), abs=1e-1) == 3751/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.boiler.costing.total_plant_cost['4.15']), abs=1e-1) == 197/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.boiler.costing.total_plant_cost['4.16']), abs=1e-1) == 1014/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.fuel_feed.costing.total_plant_cost['3.6']), abs=1e-1) == 4864/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.fuel_feed.costing.total_plant_cost['3.9']), abs=1e-1) == 522/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.fuel_feed.costing.total_plant_cost['7.3']), abs=1e-1) == 1710/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.fuel_feed.costing.total_plant_cost['7.5']), abs=1e-1) == 647/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.bfp.costing.total_plant_cost['3.1']), abs=1e-1) == 19233/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.bfp.costing.total_plant_cost['3.3']), abs=1e-1) == 6897/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.bfp.costing.total_plant_cost['3.5']), abs=1e-1) == 2366/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.bfp.costing.total_plant_cost['4.9']), abs=1e-1) == 570418/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.bfp.costing.total_plant_cost['8.4']), abs=1e-1) == 81916/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.turb.costing.total_plant_cost['8.1']), abs=1e-1) == 110166/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.condenser.costing.total_plant_cost['8.3']), abs=1e-1) == 20447/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.circulating_water.costing.total_plant_cost['9.2']),
+        abs=1e-1) == 4133/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.circulating_water.costing.total_plant_cost['9.3']),
+        abs=1e-1) == 25518/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.circulating_water.costing.total_plant_cost['9.4']),
+        abs=1e-1) == 19859/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.circulating_water.costing.total_plant_cost['9.6']),
+        abs=1e-1) == 2870/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.circulating_water.costing.total_plant_cost['9.7']),
+        abs=1e-1) == 2690/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.circulating_water.costing.total_plant_cost['14.5']),
+        abs=1e-1) == 464/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.ash_handling.costing.total_plant_cost['10.6']),
+        abs=1e-1) == 6429/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.ash_handling.costing.total_plant_cost['10.7']),
+        abs=1e-1) == 10725/1e3
+    assert pytest.approx(pyo.value(
+        m.fs.ash_handling.costing.total_plant_cost['10.9']),
+        abs=1e-1) == 2564/1e3
+
+    assert pytest.approx(pyo.value(
+        m.fs.flowsheet_cost), abs=1e-1) == 993753/1e3
+
+    return m
```

Comment:
It would be good to add unit consistency checks to these to verify that units are handled correctly.

Reply:
we aren't using units, user has to do the unit conversion before passing the scaled parameter argument
also some components will be costed as "empty" blocks